.github/workflows/push.yml

```diff
@@ -0,0 +1,65 @@
+name: Unit tests & Build Testapp
+
+on: ['push', 'pull_request']
+
+jobs:
+
+  flake8:
+    name: Flake8 tests
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout python-for-android
+      uses: actions/checkout@master
```

Comment:
Shouldn't the branch be `develop` here and the other places? Or this is not related to the p4a branch?

Reply:
Aaa...first of all, I'm not an gh-actions expert at all, but, I will try to put some light in here, so here we go:

  - `gh-actions` works based on events (In this PR we listen and react to `push` and `pull_requests` **github events**...so we kind of imitate **travis behaviour**). But, there is a lot of events that we can use, just [check this](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows) if you want to get more detailed info about events.
  - `gh-actions` is not just a name, it's a set of repos: https://github.com/actions. Those repos can be used by any other repo.

Said this, when we call the line:

  ```uses: actions/checkout@master```

What are we doing is executing the code located at: https://github.com/actions/checkout, which simply will run the proper git commands, so our workflow can access the content of our repo. The `@master` just point to the version that we use **for the specific github action** (master branch in this case but it could be a release...).


Ok, hope that this clarifies a little this particular `gh-actions` thing :wink:

Reply:
Yes it clarifies, I was exactly not sure if this branch was our or the tooling one, thanks :+1: